numpy/_core/src/multiarray/legacy_dtype_implementation.c

```diff
@@ -320,8 +320,8 @@ can_cast_fields(PyObject *field1, PyObject *field2, NPY_CASTING casting)
 
     /* Iterate over all the fields and compare for castability */
     ppos = 0;
-    while (PyDict_Next(field1, &ppos, &key, &tuple1)) {
-        if ((tuple2 = PyDict_GetItem(field2, key)) == NULL) {
+    while (PyDict_Next(field1, &ppos, &key, &tuple1)) { // noqa: borrowed-ref - manual fix needed
```

Comment:
I wasn't entirely sure about lines 323-4.

Both seem mutable, borrowed, etc. But I've kept it as `// noqa: borrowed-ref manual fix needed` for now